analysers/analyser_osmosis_powerline.py

```diff
@@ -24,28 +24,121 @@
 from .Analyser_Osmosis import Analyser_Osmosis
 from modules.Stablehash import stablehash64
 
-# Power lines nodes with their voltage as array padded up to 99 zeros
+# Power lines nodes with their voltage as array padded up to 99 zeros to cope with non-numerical values
 # Lines with no voltage get null voltage instead empty array
 sql01 = """
 CREATE TEMP TABLE power_lines_nodes AS
 SELECT
-    ways.id as wid,
-    unnest(ways.nodes) AS nid,
-    unnest(ways.nodes[2:array_length(ways.nodes,1)]) AS nid_next,
-    ways.tags->'cables' as cables,
-    ways.tags->'circuits' as circuits,
-    coalesce(ways.tags->'location','overhead') as location,
+    w.id as wid,
+    unnest('{NULL}' || w.nodes[1:array_length(w.nodes, 1) - 1]) AS nid_prec,
+    unnest(w.nodes) AS nid,
+    unnest(w.nodes[2:]) AS nid_next,
+    w.tags->'cables' AS cables,
+    coalesce((w.tags->'circuits')::integer, 1) AS circuits,
+    coalesce(w.tags->'location', 'overhead') AS location,
     voltage
 FROM
-    ways
-    JOIN LATERAL (SELECT array_agg(lpad(v, 99, '0')) FROM unnest(regexp_split_to_array(ways.tags->'voltage','; *')) AS t(v)) AS t(voltage) ON TRUE
+    ways AS w
+    JOIN LATERAL (
+        SELECT array_agg(lpad(v, 99, '0'))
+        FROM (SELECT
+            unnest(array_cat(
```

Comment:
I do not understand what you are doing here.

`length` is the length of the string. It is not about the number the voltages ? Aka length of ;-splited voltage tag, or number of ; + 1 ?

Reply:
That may be a little confusing yes: I seek for the number of the voltages in the ; list.
The only valid method I found to count the amount of ';' in the voltage string is to substract the lentgh of the voltage string minus the length of the string striped of ';' chars.
'20000;400' (9 chars) - '20000400' (8) = 1 (+1) = 2 so there are two voltages in the list.

Should I split it to an array and count the array size?

Reply:
Ok. I understand now. As you wish, add a comment or make code more readable.

Reply:
Resolve if you agree